ruby_programming/testing_with_rspec/test_driven_development.md

```diff
@@ -40,3 +40,12 @@ As usual, it depends. Still, here are some reasons we think it might be importan
 2. For a more rounded take, consider DHH's ["TDD is Dead. Long Live Testing"](http://david.heinemeierhansson.com/2014/tdd-is-dead-long-live-testing.html), which sparked controversial conversations on what TDD is, what it has become, and what it should be.
 
 </div>
+
+### Knowledge Check
+
+This section contains questions for you to check your understanding of this lesson. If you’re having trouble answering the questions below on your own, review the material above to find the answer.
+
+- <a class="knowledge-check-link" href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/test-driven-development#what-is-test-driven-development">What is the "red-green-refactor" cycle?</a>
+- <a class="knowledge-check-link" href="https://thoughtbot.com/blog/rails-test-types-and-the-testing-pyramid#unit-tests">What is the difference between unit and integration tests?</a>
```

Comment:
I see that this is where the link on line 25 now goes, can you update it there?

Reply:
Yes, I will do that.

Reply:
How to I update the link? Here on GitHub or on my local repository? Do I need a new branch?

Reply:
Do I have to make a new PR?

Reply:
No don't make a new PR, you can make changes on your local copy of the repo and push them to the same branch that you're doing the PR from, or you can make the edits from GitHub directly, again just make sure the changes go to your `feature/knowledgeChecks` branch of your fork.

Reply:
I am a bit confused now. So how should the link be:

href="https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby-programming/lessons/test-driven-development#what-is-test-driven-development"

or

href="#what-is-test-driven-development"
?

Reply:
I decided that it had to be the second option. 
I fixed it and hope it is OK. 
Thanks for your help. I learned a lot!

Reply:
Looks like you addressed the first change I requested, but the link on line 25 is still not the same URL used in your Knowledge Check on line 49, can you update the line 25 link to match your line 49 link?

If I'm being unclear here please let me know.